muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/WorkspaceMaterialController.java

```diff
@@ -743,24 +743,16 @@ public Long getMaterialCountByWorkspaceAndAssignmentType(WorkspaceEntity workspa
   public List<WorkspaceMaterial> listVisibleWorkspaceMaterialsByAssignmentType(WorkspaceEntity workspaceEntity, WorkspaceMaterialAssignmentType assignmentType) {
     final List<WorkspaceNode> folders = listVisibleWorkspaceFolders(workspaceEntity);
     List<WorkspaceMaterial> workspaceMaterials = workspaceMaterialDAO.listByHiddenAndAssignmentTypeAndParents(Boolean.FALSE, assignmentType, folders);
-
     Collections.sort(workspaceMaterials, new Comparator<WorkspaceMaterial>() {
-      
       @Override
       public int compare(WorkspaceMaterial o1, WorkspaceMaterial o2) {
-        int result = getParentIndex(o1.getParent()) - getParentIndex(o2.getParent());
+        int result = o1.getParent().getOrderNumber() - o2.getParent().getOrderNumber();
         if (result == 0) {
           result = o1.getOrderNumber().compareTo(o2.getOrderNumber());
```

Comment:
This is probably redundant, compareTo does the same (kind of) comparison between the ordernumbers. Afaik the magnitude doesn't matter, only the -1/0/1 output. So maybe just 
o1.getOrderNumber().compareTo(o2.getOrderNumber()); for simplicity?